tests/nada-tests/tests/broadcasting_vec.yaml

```diff
@@ -1,111 +1,57 @@
 program: broadcasting_vec
 inputs:
-  A_0_0_0:
-    SecretInteger: '3'
-  A_0_1_0:
-    SecretInteger: '3'
-  A_0_0_2:
-    SecretInteger: '3'
-  A_2_1_1:
-    SecretInteger: '3'
-  A_2_2_1:
-    SecretInteger: '3'
-  A_1_1_2:
-    SecretInteger: '3'
-  A_1_2_0:
-    SecretInteger: '3'
-  A_0_2_0:
-    SecretInteger: '3'
-  A_1_1_0:
-    SecretInteger: '3'
-  A_1_0_1:
-    SecretInteger: '3'
-  A_0_0_1:
-    SecretInteger: '3'
-  A_0_1_2:
-    SecretInteger: '3'
-  A_1_2_1:
-    SecretInteger: '3'
-  A_2_1_0:
-    SecretInteger: '3'
-  A_0_2_1:
-    SecretInteger: '3'
-  A_2_0_0:
-    SecretInteger: '3'
-  A_2_1_2:
-    SecretInteger: '3'
-  A_1_1_1:
-    SecretInteger: '3'
-  A_0_2_2:
-    SecretInteger: '3'
-  A_1_0_0:
-    SecretInteger: '3'
-  A_1_0_2:
-    SecretInteger: '3'
-  A_2_0_1:
-    SecretInteger: '3'
-  A_2_0_2:
-    SecretInteger: '3'
-  A_2_2_0:
-    SecretInteger: '3'
-  A_2_2_2:
-    SecretInteger: '3'
-  A_1_2_2:
-    SecretInteger: '3'
-  A_0_1_1:
-    SecretInteger: '3'
+  A_0_0_0: 3
```

Comment:
Looking at this, it would actually be quite nice if we could put a (possibly nested) list in these YAMLs
So something like `A: [[1, 2, 3], [4, 5, 6], ...]`

Reply:
WIth the new testing infrastructure, I would like to deprecate the use of these files and creating programmatic tests that check for all these behaviours. In such case, we can directly input lists and forget about filling up ginormous files.